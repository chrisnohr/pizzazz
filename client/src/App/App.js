import React from 'react';
import Message from '../Message/Message';
import GameDnd from '../GameDnd/GameDnd';
import Inputs from '../Inputs/Inputs';
import ProgressBar from '../ProgressBar/ProgressBar';
import Definitions from '../Definitions/Definitions';
import AppWrapper from './AppStyled';

const App = () => (
  <AppWrapper>
    <Message />
    <Inputs />
    <GameDnd />
    <ProgressBar />
    <Definitions />

  </AppWrapper>
);


export default App;
