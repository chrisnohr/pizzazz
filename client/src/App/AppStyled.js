import styled from 'styled-components';
import { spacer, centerH } from '../styleConstants';

const AppStyled = styled.div`
  ${centerH};
  text-align: center;
  max-width: 500px;
  margin-top: 18vw;
  box-sizing: border-box;
  font-family: Roboto, sans-serif;
  @media screen and (min-width: 500px) {
    margin-top: ${spacer};
  }
`;

export default AppStyled;
