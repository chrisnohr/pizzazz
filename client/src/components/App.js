// @flow
/* eslint react/prefer-stateless-function: off */
// react dnd warning when using stateless component
import React, { Component } from 'react';
import { DragDropContext } from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import TouchBackend from 'react-dnd-touch-backend';
import MultiBackend, { TouchTransition } from 'react-dnd-multi-backend';
import RandomLetters from './RandomLetters';
import Interface from './Interface';
import Message from './Message';
import ScrabbleBoard from './ScrabbleBoard';
import DragLayerTile from './DragLayerTile';
import AppWrapper from './styled/AppStyled';
import type { Node } from 'react';


const HTML5toTouch = {
  backends: [
    {
      backend: HTML5Backend,
    },
    {
      backend: TouchBackend({ enableMouseEvents: true }),
      preview: true,
      transition: TouchTransition,
    },
  ],
};


export class App extends Component<{}> {
  render(): Node {
    return (
      <AppWrapper>
        <Message />
        <RandomLetters />
        <ScrabbleBoard />
        <Interface />
        <DragLayerTile />
      </AppWrapper>
    );
  }
}


export default DragDropContext(MultiBackend(HTML5toTouch))(App);
