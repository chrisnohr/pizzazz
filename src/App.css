.App {
  text-align: center;
}

.is-valid {
  border-bottom: 4px solid green;
}
