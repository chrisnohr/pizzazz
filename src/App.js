import React, { Component } from 'react'
import RandomLetters from './RandomLetters'
import MakeRandomLettersButton from './MakeLettersButton'
import ValidWord from './ValidWord'
import { DragDropContext } from 'react-dnd'
import HTML5Backend from 'react-dnd-html5-backend'
import { replaceLetterAction } from './actions'

class App extends Component {
  render() {
    return (
      <div className="text-center m-5">
        <RandomLetters replaceLetter={ replaceLetterAction } />
        <ValidWord replaceLetter={ replaceLetterAction } />
        <MakeRandomLettersButton />
      </div>
    )
  }
}

export default DragDropContext(HTML5Backend)(App);
