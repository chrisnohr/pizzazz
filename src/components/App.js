import React, { Component } from 'react'
import styled from 'styled-components'
import RandomLettersHTML from './RandomLettersHTML'
import CountdownContainer from './CountdownContainer'
import GameInterfaceBottom from './GameInterfaceBottom'
import GameInterfaceTop from './GameInterfaceTop'
import ValidWordHTML from './ValidWordHTML'
import CountdownPizzazz from './CountdownPizzazz'
import { DragDropContext } from 'react-dnd'
import HTML5Backend from 'react-dnd-html5-backend'

const AppWrapper = styled.div`
  height: 100vh;
  background-color: #F9F8F2;
  text-align: center;
  padding: 3rem 0;
`;

class App extends Component {
  render() {
    return (
      <AppWrapper>
        <GameInterfaceTop />
        <RandomLettersHTML />
        <ValidWordHTML />
        <GameInterfaceBottom />
        <CountdownContainer />
      </AppWrapper>
    )
  }
}

export default DragDropContext(HTML5Backend)(App);
