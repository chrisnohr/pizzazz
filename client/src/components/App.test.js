import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('<App />', () => {
  it('renders an element with 5 children', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.props().children).toHaveLength(5)
  })
  it('matches the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })
})
