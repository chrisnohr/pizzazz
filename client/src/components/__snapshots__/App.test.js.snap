// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<styled.div>
  <Connect(Message) />
  <Connect(RandomLetters) />
  <Connect(ScrabbleBoard) />
  <Connect(Interface) />
  <DragLayer(DragLayerTile) />
</styled.div>
`;
